R/unnecessary_nesting_linter.R

```diff
@@ -232,7 +232,9 @@ unnecessary_nesting_linter <- function(
       # catch if (cond) if (other_cond) { ... }
       "following-sibling::expr[IF and not(ELSE)]",
       # catch if (cond) { if (other_cond) { ... } }
-      "following-sibling::expr[OP-LEFT-BRACE and count(expr) = 1]/expr[IF and not(ELSE)]"
+      #   count(*): only OP-LEFT-BRACE, one <expr>, and OP-RIGHT-BRACE.
+      #             Note that third node could be <expr_or_assign_or_help>.
+      "following-sibling::expr[OP-LEFT-BRACE and count(*) = 3]/expr[IF and not(ELSE)]"
```

Comment:
Does this play well with `if (A) if (B) c`, i.e. without braces?

Reply:
Yes:

https://github.com/r-lib/lintr/blob/15b75d80dfc932bbd32fdf08a0d7ed9f5b38a540/tests/testthat/test-unnecessary_nesting_linter.R#L644

If you see any other missing cases in there, we can add their tests as a follow-up